### PR TITLE
/:kind element API

### PR DESCRIPTION
**Pull request by maasencioh**
Not finished, so please still don't merge yet!

### CODE REVIEW

**Comment by targos** [comment]
I'll review the code when you say it is ready, but the middleware should be in https://github.com/cheminfo/hds-api/blob/master/src/routes/entry.js, not index ;)

File: src/routes/entry.js [hunk 1]
```diff
@@ -64,4 +66,15 @@ module.exports = function (hds) {
         }
     }
 
+    function* createEntry(kind) {
+        var data = this.query;
+        hds.Kind.create(kind, data).save().then(function (value) {
```
Comment (targos) [inline src/routes/entry.js]: here use yield

Comment (targos) [inline src/routes/entry.js]:
``` js
try {
  yield hds.Kind.create(kind, data).save();
} catch (e) {
  this.hds_jsonError(400, err);
}

this.body = {
  status: 'created',
  entryID: value._id
};
```

File: src/routes/entry.js [hunk 2]
```diff
@@ -15,13 +15,15 @@ module.exports = function (hds) {
 
     var router = new Router();
 
+    router.post('/:kind', createEntry);
```
Comment (targos) [inline src/routes/entry.js]: add a `checkKind` step

File: src/routes/entry.js [hunk 3]
```diff
@@ -52,10 +73,63 @@ module.exports = function (hds) {
         this.body = entry;
     }
 
-    function*getAttachment() {
+    function* createEntry() {
+        var data = this.request.body;
+        try {
+            var value = yield hds.Entry.create(this.params.kind, data, {owner: 'test@cheminfo.org'}).save();
```
Comment (maasencioh) [inline src/routes/entry.js]: @targos @lpatiny @andcastillo when we did the document we forgot to specify the owner part, and thats mandatory for the creation of an entry, any idea of how to solve it?

File: src/routes/entry.js [hunk 4]
```diff
@@ -52,10 +63,66 @@ module.exports = function (hds) {
...
+
+    function* queryEntry() {
+        try {
+            var data = this.request.body;
+            var from = data.from || 0;
+            var limit = data.limit || 20;
+            var entries = yield this.state.hds_kind
+                .find(data.query)
+                .skip(from)
+                .limit(limit - from)
```
Comment (maasencioh) [inline src/routes/entry.js]: checking the doc I'm still not sure if this is the expected value for `.limit()`

File: src/routes/entry.js (hunk 3 above)
Comment (targos) [inline src/routes/entry.js]: `{owner: this.state.user.email}`

Comment (targos) [inline src/routes/entry.js]: put this somewhere in the server, before hds api is called :

```
this.state.user = {
  email: 'test@test.com'
}
```

**Pull request merged**